app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -343,21 +345,29 @@ private static PokemonNameCorrector initCorrectorFromPrefs(PokeInfoCalculator po
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        if (intent != null && intent.hasExtra(KEY_TRAINER_LEVEL)) {
-            trainerLevel = intent.getIntExtra(KEY_TRAINER_LEVEL, 1);
-            statusBarHeight = intent.getIntExtra(KEY_STATUS_BAR_HEIGHT, 0);
-            batterySaver = intent.getBooleanExtra(KEY_BATTERY_SAVER, false);
-            makeNotification(this);
-            createInfoLayout();
-            createIVButton();
-            createArcPointer();
-            createArcAdjuster();
-            /* Assumes MainActivity initialized ScreenGrabber before starting this service. */
-            if (!batterySaver) {
-                screen = ScreenGrabber.getInstance();
-                startPeriodicScreenScan();
-            } else {
-                screenShotHelper = ScreenShotHelper.start(Pokefly.this);
+        if (intent != null) {
```

Comment:
I know about the unnecessary diffs, but the code needed to be refractored.
The null check on the Intent is needed in both cases so embraces the two other checks.

I use CONSTANT.equals(...) as many Google employees do since it avoids NullPointerExceptions: the constant will never be null while the compared value could (since it's out of our control, it comes from a system broadcast)!
Anyway it could be reversed and tested: if in the normal lifecycle of the service null action intents are never received for me it's ok to change!